webviz_subsurface/plugins/_group_tree/group_tree.py

```diff
@@ -0,0 +1,215 @@
+                    "WWIR*",
+                    "WGIR*",
+                    "WTHP*",
+                    "WBHP*",
+                    "WMCTL*",
+                    "WSTAT*",
+                ],
+            )
+        )
+        smry = self.emodel.get_or_load_smry_cached()
+        gruptree = read_gruptree_files(self.ens_paths, self.gruptree_file)
+        smry["DATE"] = pd.to_datetime(smry["DATE"]).dt.date
+        gruptree["DATE"] = pd.to_datetime(gruptree["DATE"]).dt.date
+
+        self.grouptreedata = GroupTreeData(smry, gruptree)
+
+        self.set_callbacks(app)
+
+    def add_webvizstore(self) -> List[Tuple[Callable, List[Dict]]]:
+        functions: List[Tuple[Callable, List[Dict]]] = self.emodel.webvizstore
+        functions.append(
+            (
+                read_gruptree_files,
+                [{"ens_paths": self.ens_paths, "gruptree_file": self.gruptree_file}],
+            )
+        )
+        return functions
+
+    @property
+    def tour_steps(self) -> List[dict]:
+        return [
+            {
+                "id": self.uuid("layout"),
+                "content": "Dashboard vizualizing Eclipse network tree.",
+            },
+            {
+                "id": self.uuid("selections_layout"),
+                "content": "Menu for selecting ensemble and other options.",
+            },
+            {
+                "id": self.uuid("filters_layout"),
+                "content": "Menu for filtering options.",
+            },
+            {
+                "id": self.uuid("grouptree_wrapper"),
+                "content": "Vizualisation of network tree.",
+            },
+        ]
+
+    @property
+    def layout(self) -> html.Div:
+        return html.Div(
+            children=[
+                # clientside_stores(get_uuid=self.uuid),
+                main_view(get_uuid=self.uuid, ensembles=self.ensembles),
+            ],
+        )
+
+    def set_callbacks(self, app: dash.Dash) -> None:
+        controllers(app=app, get_uuid=self.uuid, grouptreedata=self.grouptreedata)
+
+
+@webvizstore
+def read_gruptree_files(ens_paths: Dict[str, str], gruptree_file: str) -> pd.DataFrame:
+    """Searches for gruptree files on the scratch disk. These
+    files can be exported in the FMU workflow using the ECL2CSV
+    forward job with subcommand gruptree.
+    """
+    df = pd.DataFrame()
+    for ens_name, ens_path in ens_paths.items():
+        df_ens = read_ensemble_gruptree(ens_name, ens_path, gruptree_file)
+        df_ens["ENSEMBLE"] = ens_name
+        df = pd.concat([df, df_ens])
+    return df.where(pd.notnull(df), None)
+
+
+def read_ensemble_gruptree(
+    ens_name: str, ens_path: str, gruptree_file: str
+) -> pd.DataFrame:
+    """Reads the gruptree file for and ensemble.
+
+    If BRANPROP is found in the KEYWORD column, then GRUPTREE rows
+    are filtered out.
+
+    If the trees are equal in every realization, only one realization is kept.
+    """
+    ens = ScratchEnsemble("ens", ens_path)
```

Comment:
the ensemble needs to be filtered on `OK` for consistency, but I think it might be better to use the ensembles through the EnsembleSetModel in `self.emodel` should already contain ensembles in the internal property `_ensembles` (should maybe be exposed?), and the filtering is also handled there. 
Maybe better to use that instead of introducing a new place where we initialize ensembles (mainly to keep control of ensemble initializations and avoid the classic "missing OK filter error" that we encounter a bit too often 🙈).

Reply:
I solved this one by importing the `scratch_ensemble` function from `_datatinput/fmu_input`. That function has the OK-filter and is cached so should solve the memory issue.